### PR TITLE
[RA2 Ch3] Reorder 3.2.1.x sections and TOC

### LINKED ISSUE #2140: [RA2 Ch3] Reorder 3.2.1.x sections and TOC
Reorder [3.2.1.x](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter03.md#3.2.1.1) sections and TOC so it is sorted by what is always needed (3.2.1.1 Container Runtime Services) and so fits with flow of other parts coming to 3.2.1.

### PR DESCRIPTION
**Pull request by petorre**
Fixes #2140 .

Reorder 3.2.1.x https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter03.md#3.2.1 sections and TOC, so it is sorted by starting with what is always needed (3.2.1.1 Container Runtime Services) and better fits with flow of other parts.

### CODE REVIEW

**Review by tomkivlin — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 1]
```diff
@@ -206,6 +175,42 @@ which provides the isolation of Operating System kernels.
 The architecture must support a way to isolate the compute resources of the
 infrastructure itself from the workloads compute resources.
 
+
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:178]: The updated ToC is:
- 3.2.1.1 Container Runtime Services
- 3.2.1.2 CPU Management
- 3.2.1.3 Memory and Huge Pages Resources Management
- 3.2.1.4 Hardware Topology Management
- 3.2.1.5 Node Feature Discovery
- 3.2.1.6 Device Plugin Framework
- 3.2.1.7 Hardware Acceleration
- 3.2.1.8 Scheduling Pods with Non-resilient Applications

The numbering below is inconsistent with the new ToC.

Comment (petorre) [reply doc/ref_arch/kubernetes/chapters/chapter03.md:178]: > The numbering below is inconsistent with the new ToC.

Yes, numbering for sections in text is coming in #2147 , #2148 , #2150 , #2151 , #2152 , #2153 and #2154 .

**Review by ulikleber — APPROVED** [review]

**Comment by tomkivlin** [comment]
@rgstori could you review/approve this one too please?

**Comment by tomkivlin** [comment]
@petorre could you resolve the conflicts in this one too please?

**Review by rgstori — APPROVED** [review]

**Comment by petorre** [comment]
> @petorre could you resolve the conflicts in this one too please?

@tomkivlin : Done.

**Pull request merged**